src/cpp/ports.cpp

```diff
@@ -508,8 +508,10 @@ Ports::Ports(std::initializer_list<Port> l)
 
 Ports::~Ports()
 {
-    delete []impl->enump;
-    delete impl;
+    if(impl) {
```

Comment:
I would just add a Port_Matcher::~Port_Matcher(void) method and then you can just have "delete impl". Let C++ handle the implicit if(x) get_rid_of(x) logic for you.